### ISSUE TITLE
Spring preloading and cucumber queue mode

### ISSUE THREAD

**Issue by andrcuns** [KnapsackPro/knapsack_pro-ruby#97]
Hi,

I have been updating our cucumber tests to the new queue mode and encountered one issue.
Because it is executing tests in chunks, we are loosing quite a bit of time on initial rails initialisation each time new chunk is started. Having ability to use spring would help with this issue, but doesn't seem it is possible at the moment with the queue runner.

I found there are 2 ways to implement this:
1. Having some sort of env var like `KNAPSACK_PRO_CUCUMBER_QUEUE_SPRING_ENABLED` which would transform [command](https://github.com/KnapsackPro/knapsack_pro-ruby/blob/05daca4e07f5bca5e5d924cc8592c698a094df6b/lib/knapsack_pro/runners/queue/cucumber_runner.rb#L82) to a `bundle exec spring cucumber`.
2. Having env var similar to cucumber rake task option `binary`, which would allow to replace executed cucumber binary with a springified binstub.

WDYT? Maybe there are some other ways?

**Comment by ArturT** [comment]
I'm wondering if we need to consider support for something else than just spring.
I recall there is some new modern gem that's alternative to spring but I forgot the name of it and can't google it.

Regarding how this could be done. Maybe we could have a `prefix` and you could define it in env `KNAPSACK_PRO_CUCUMBER_QUEUE_MODE_PREFIX` then it could be `spring` or `bin/spring`, or if you use something else then you could change the prefix.

```ruby
prefix = ENV['KNAPSACK_PRO_CUCUMBER_QUEUE_MODE_PREFIX'] || 'bundle exec'
cmd = %Q[#{prefix} cucumber #{args} --require #{runner.test_dir} -- #{stringify_test_file_paths}]
```

**Comment by andrcuns** [comment]
Yeah, that works too if You want to support some other alternatives.
Quick googling showed me one other alternative written by shopify - [bootsnap](https://github.com/Shopify/bootsnap), looks like that one would work without any changes to exec command.
PREFIX could be also helpful if by some chance someone doesn't have bundler in path or something similar. It's up to You how to implement it, either way would work for our use case.

**Comment by ArturT** [comment]
@andrcuns I've created PR for that https://github.com/KnapsackPro/knapsack_pro-ruby/pull/98

I'm going to prepare documentation and update FAQ and release a new gem version hopefully tomorrow. In the meantime, you can test it out from the branch.

**Comment by andrcuns** [comment]
Tried running from the branch, worked as expected 👍

**Comment by ArturT** [comment]
@andrcuns I've released a new version of knapsack_pro gem 1.17.0. Please update it.

I also updated README, installation guide and added info to FAQ https://knapsackpro.com/faq/question/how-to-use-spring-gem-to-run-cucumber-tests-faster-in-queue-mode
